src/compiler/checker.ts

```diff
@@ -3687,12 +3687,18 @@ namespace ts {
                         return cloneTypeAsModuleType(symbol, defaultOnlyType, referenceParent);
                     }
 
-                    if (getESModuleInterop(compilerOptions)) {
+                    const targetFile = moduleSymbol?.declarations?.find(isSourceFile);

```

Comment:
What about ambient module declarations that use `export =`? They can only be describing CJS modules, right? Shouldn’t they ideally have the same behavior as source files that were definitely CJS?

Reply:
I mean... So technically we'd probably wanna consider all ambient modules as cjs modules, for compat's sake, problem being there's no ambient module syntax for a definitively esm ambient module, or for having both module types existing at the same name. It's a bit of a thorny issue that I've been hoping to not need to fix until it was truly required, since the reasons to use ambient modules over modules on disk are pretty small.

Reply:
Module declaration assertions when? 😄